Dockerfile

```diff
@@ -94,3 +94,43 @@ WORKDIR /workspace
 COPY --from=builder /renv /renv
 # this will ensure the renv is activated by default
 RUN echo 'source("/renv/renv/activate.R")' >> /etc/R/Rprofile.site
+
+#################################################
+#
+# Add rstudio-server to r image - creating rstudio image
+FROM r as rstudio
+
+COPY rstudio-dependencies.txt /root/rstudio-dependencies.txt
+COPY rstudio-entrypoint.sh /usr/local/bin/rstudio-entrypoint.sh
+
+# Install rstudio-server (and a few dependencies)
+RUN --mount=type=cache,target=/var/cache/apt /root/docker-apt-install.sh /root/rstudio-dependencies.txt &&\
+    wget -q -O rstudio-server.deb https://download2.rstudio.org/server/focal/amd64/rstudio-server-2024.09.0-375-amd64.deb &&\
```

Comment:
Ok, so some possible docker improvements here

1) we could parameterise the Rstudio version with docker build ARG, and use that to construct the url (or the build arg could be the entire url). This way its easy to test/upgrade to a new version of rstudio by changing some config (e.g. in the justfile) rather than editing the Dockerfile

2) Rather than deleting, and to save time in development, it might be worth downloading it to host cache /var/cache/apt, and only downloading it if not present e.g. `test -f /var/cache/apt/rstudio.deb || wget ...`. Then we don't `rm` it either.

3) To download things in other Dockerfile's we use a simple apt downloader rather than installing curl or wget. e.g.
```
/usr/lib/apt/apt-helper download-file https://download2.rstudio.org/server/focal/amd64/rstudio-server-2024.09.0-375-amd64.deb /var/cache/apt/rstudio-server.deb
```

Reply:
I've made all 3 changes. The `ARG` is `RSTUDIO_DEB`. It's default value is set in the *justfile*.